outdoorsmanship.lic

```diff
@@ -9,21 +9,44 @@ class Outdoorsmanship
   include DRCI
 
   def initialize
+    arg_definitions = [
+      [
+        { name: 'mindstates', regex: /\d+/, optional: true, description: 'Number of mindstates or collection attempts to perform before exiting. Defaults to 3' },
+        { name: 'collect_item', regex: /\w+/i, optional: true, description: 'Item to collect. Will use forage_item if not set' }
+      ]
+    ]
+    args = parse_args(arg_definitions)
+
     settings = get_settings
-    @forage_item = settings.forage_item
+    
+    @targetxp = 3
+    @targetxp = args.mindstates.to_i if args.mindstates
+    
+    if args.collect_item
+      @forage_item = args.collect_item
+    else
+      @forage_item = settings.forage_item
+    end
+
+    start_exp = DRSkill.getxp('Outdoorsmanship')
+    @end_exp = [start_exp + @targetxp, 29].min
 
     train_outdoorsmanship
   end
 
   def train_outdoorsmanship
-    if DRSkill.getrank('Outdoorsmanship') < 20
-      forage?('rock')
-      dispose_trash 'rock'
-    else
-      item = @forage_item
-      collect(item)
-      waitrt?
-      kick_pile? unless kick_pile?("#{item}s")
+    attempt = 0   # Failsafe counter to prevent an infinite loop if item isn't collectable in the room
```

Comment:
💡 I'm not opposed to this change, but I feel like `collect(...)` should maybe be a predicate and handle this.

Reply:
I thought about that, but figured that would mean the method name would need to change to `collect?`, and I worry about breaking other scripts, especially if people have stuff that's not in the repo that uses that method

Reply:
If you were to do this, my recommendation would be to have one PR that _only adds_ `collect?` (and could even be mostly a copy/paste of `collect`) and then another PR a few days later that converts all the scripts to use `collect?` instead of `collect` (and deletes `collect`).

This gives people running multiple characters time to have re-run `common` before the consumer script changes.